hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/utils/HoodieHiveUtils.java

```diff
@@ -71,7 +71,7 @@
   public static final String DEFAULT_SCAN_MODE = SNAPSHOT_SCAN_MODE;
   public static final int DEFAULT_MAX_COMMITS = 1;
   public static final int MAX_COMMIT_ALL = -1;
-  public static final int DEFAULT_LEVELS_TO_BASEPATH = 3;
+  public static final int DEFAULT_LEVELS_TO_BASEPATH = 0;
```

Comment:
@dongkelun this is just a default value. this should not be the root cause. The main problem should be somewhere in the code the actual levels blindly using the default and not overwritten by the actual one. Can you trace down the usage and figure out the misuse?

Reply:
Hello, I know this is only a default value. There is only one place to use it globally, which is the `HoodieInputFormatUtils.getTableMetaClientForBasePathUnchecked` method. Its logic is to use `HoodiePartitionMetadata.getPartitionDepth` get the actual levels, but an empty table without data will not  create `.hoodie_partition_metadata` file,so  It can't get levels, then an exception will be thrown. I think  either change the default value or create `.hoodie_partition_metadata`  file when creating an empty table , and `.hoodie_partition_metadata`  cannot be deleted when `truncate table`.
I'm not sure if there's any better way. What do you think?